### PR TITLE
Arithmetic operations for LinearDelayModel

### PR DESCRIPTION
**Pull request by pmli**
refs #591

### CODE REVIEW

File: src/pymor/models/iosys.py [hunk 1]
```diff
@@ -1500,6 +1500,161 @@ def __init__(self, A, Ad, tau, B, C, D=None, E=None, cont_time=True,
...
+        return self.with_(E=E, A=A, Ad=Ad, tau=tau, B=B, C=C, D=D)
+
+    def __radd__(self, other):
+        """Add to an |LTIModel| or a |SecondOrderModel|."""
+        if isinstance(other, LTIModel):
+            return self + other
+        elif isinstance(other, SecondOrderModel):
+            return self + other.to_lti()
+        else:
+            raise NotImplementedError
```
Comment (sdrave) [inline src/pymor/models/iosys.py]: Shouldn't this be `return NotImplemented`?

File: src/pymor/models/iosys.py [hunk 2]
```diff
@@ -1500,6 +1500,161 @@ def __init__(self, A, Ad, tau, B, C, D=None, E=None, cont_time=True,
...
+        else:
+            D = self.D - other.D
+        return self.with_(E=E, A=A, Ad=Ad, tau=tau, B=B, C=C, D=D)
+
+    def __rsub__(self, other):
+        """Subtract from an |LTIModel| or a |SecondOrderModel|."""
+        if isinstance(other, (LTIModel, SecondOrderModel)):
+            return -(self - other)
+        else:
+            raise NotImplementedError
```
Comment (sdrave) [inline src/pymor/models/iosys.py]: See above.

File: src/pymor/models/iosys.py [hunk 3]
```diff
@@ -1500,6 +1500,161 @@ def __init__(self, A, Ad, tau, B, C, D=None, E=None, cont_time=True,
...
+            A = BlockOperator([[other.A, other.B @ self.C],
+                               [None, self.A]])
+            Ad = tuple(BlockDiagonalOperator([ZeroOperator(other.solution_space, other.solution_space), op])
+                       for op in self.Ad)
+            B = BlockColumnOperator([other.B @ self.D, self.B])
+            C = BlockRowOperator([other.C, other.D @ self.C])
+            D = other.D @ self.D
+            return self.with_(E=E, A=A, Ad=Ad, B=B, C=C, D=D)
+        else:
+            raise NotImplementedError
```
Comment (sdrave) [inline src/pymor/models/iosys.py]: See above.

File: src/pymor/models/iosys.py (hunk 1 above)
Comment (pmli) [reply src/pymor/models/iosys.py]: Right. Fixed it now also in `SecondOrderModel`.

**Pull request merged**